### ISSUE TITLE
Question on nested tables with slim

### ISSUE THREAD

**Issue by brucetrask** [unclebob/fitnesse#1012]
Is it possible to use more complicated tables than hash tables with Slim Fixtures?
For example I would like to do what is possible with FitLibrary such as
![domainobjectfixture](https://cloud.githubusercontent.com/assets/10885119/21885951/196b59c4-d8ba-11e6-9aa6-6e6b6c07ac5c.PNG)
and in the end, with FitLibrary, I can do arbitrary tables within tables using its DoTraverse capabilities. But I would like to use more complex tables with Slim scripts and scenarios.
Is that possible relatively easily with Slim?

**Comment by fhoeben** [comment]
I use Slim with nested maps/tables in combination with script (and scenario) tables. 

As a caveat: I don't define the maps in wiki directly (using the hash widget) but instead use a custom fixture that allows me to define these maps and assign them to 'slim variables/symbols' that I pass to my fixtures via script (and scenario) tables. The Java fixtures classes then receive the values as parameters (as nested Java maps).
You can see [an example in the acceptance tests of my fixtures](http://fhoeben.github.io/hsac-fitnesse-fixtures/acceptance-test-results/HsacAcceptanceTests.SlimTests.UtilityFixtures.MapFixtureTest.html).

**Comment by brucetrask** [comment]
Hi fhoeben, Thanks for your response and it looks like you have done some great work with the Maps. My requirements are  bit different in that I need to send an embedded decision table that is not a map but rather a set of rows and columns that are more than two columns wide. FitLibrary provides some elegant solutions for this but I wanted to do the same thing with Slim and so was wondering if it supported that. In my original post there is an example table in that that has three columns: name, phone, address and that is a good example of what i am trying to do with Slim. Not sure though if that is supported in any way.

**Comment by fhoeben** [comment]
Sorry if it can't help you. Looking at your original example it looked like a 'script fixture' supporting the creation of a book might be an option. I thought creating a (nested) map with all the book's properties and then a method that created based on that map might be an option. (The fixtures I use do support nested maps with more than 2 columns, and lists of them.)
But indeed there is no support for a nested decision table, as far as I know.

**Comment by brucetrask** [comment]
Ok. Appreciate your response. I think I will try a simple example  of passing a decision table in to a script and see if can getting working with Slim via the source. Thanks.

**Comment by six42** [comment]
Hi Brucetrask,

Can you publish the fixture source code of your example?
What is the expected  structure of the java object created by the create book fixture?

**Comment by brucetrask** [comment]
Yes. This one is straight out of the fitlibrary distribution. https://sourceforge.net/p/fitlibrary/fitlibrary/ci/6e2fd68caab2f02de2d8c7d77899562c496a4b1d/tree/src/fitlibrary/specify/AggregateDomainObject.java
